### ISSUE TITLE
solders.SerdeJSONError: data did not match any variant of untagged enum IdlAccountItem at line 1 column 391

### ISSUE THREAD

**Issue by siyangyuandegit** [kevinheavey/anchorpy#149]
How can i fix the bug?

**Comment by siyangyuandegit** [comment]
from anchorpy import Program, Provider, Wallet, Idl
from solana.rpc.async_api import AsyncClient
from solders.keypair import Keypair
import json


async def load_program():

    connection = AsyncClient("https://api.mainnet-beta.solana.com")
    
    wallet = Wallet(Keypair())
    
    provider = Provider(connection, wallet)
    
    with open("idls/pumpfun.json") as f:
        raw_idl = json.load(f)
    
    idl = Idl.from_json(json.dumps(raw_idl))
    print
    program = Program(
        idl,
        "6EF8rrecthR5Dkzon8Nwu78hRvfCKubJ14M5uBEwF6P",
        provider
    )
    
    return program

async def main():
    program = await load_program()
    print("Program loaded successfully!")
    print("Program ID:", program.program_id)
    print("Available instructions:", program.instruction_names)
    print("Available events:", [e.name for e in program.events])

if __name__ == "__main__":
    import asyncio
    asyncio.run(main())
  I want to init the program, but i can't

**Comment by Hari-hari-dev** [comment]
@siyangyuandegit Reduce you Anchor version to 0.20.1 as well.

**Comment by Hari-hari-dev** [comment]
> @Hari-hari-dev wdym "reduce" to 0.20.1, the latest version is 0.20.1? https://pypi.org/project/anchorpy/#history
> 
> UPDATE: nvm, I see which hash the anchor folder is pointed to ..
> 
> Note that the issue could also be this: #147

Yeah, #147 explains it. Anchor 0.27.0 works too. I think up to 0.29.0. 0.30.1 gives enum errors.

**Comment by quantitative-technologies** [comment]
How do I "reduce my Anchor version"? `anchorpy` doesn't install an anchor library does it?

**Comment by JamesGalt** [comment]
While we are waiting for #147 here a script to convert to "legacy format" 

[convert_idl.py](https://github.com/user-attachments/files/24375519/convert_idl.py)

note: some fields in the new format aren't supported in the legacy one, they are dropped.

**Comment by blinkenstephanie** [comment]
`IdlAccountItem` untagged-enum mismatches usually mean the IDL JSON is newer (or older) than solders/anchorpy expect — pin IDL schema version, or normalize `accounts` entries (string vs `{name,writable,signer}` vs nested tree) before parse. Golden-fixture the current Pump AMM IDL. Soft: APEX local PumpSwap toolkit — https://github.com/blinkenstephanie/apex-solana-pumpswap-launcher · @suntzuson
